### PR TITLE
Ram 2500 + 3500 HD support

### PR DESCRIPTION
**Pull request by realfast**
This implements Ram HD trucks into the standard Chrysler brand.

### CODE REVIEW

File: selfdrive/car/torque_data/override.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ COMMA BODY: [.nan, 1000, .nan]
 
 # Totally new cars
 KIA EV6 2022: [3.5, 2.5, 0.0]
-RAM 1500 5TH GEN: [2.0, 2.0, 0.0]
-SUBARU OUTBACK 6TH GEN: [2.3, 2.3, 0.11]
+RAM 1500 5TH GEN: [2.68, 2.68, 0.0]
```
Comment (adeebshihadeh) [inline selfdrive/car/torque_data/override.yaml:24]: revert this
```suggestion
RAM 1500 5TH GEN: [2.0, 2.0, 0.0]
```

File: selfdrive/car/chrysler/values.py [hunk 2]
```diff
@@ -121,60 +131,108 @@ class ChryslerCarInfo(CarInfo):
...
+      b'68320950AM',
       b'68454268AB',
+      b'68475160AE',
       b'68475160AG',
       b'04672892AB',
-      b'68475160AE',
     ],
     (Ecu.eps, 0x75A, None): [
-      b'68273275AG',
+      b'68273275AG', #S0
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:164]: can you remove these other FW versions and make a separate PR for them?

File: selfdrive/car/chrysler/values.py [hunk 3]
```diff
@@ -22,23 +22,32 @@ class CAR:
 
   # Ram
   RAM_1500 = "RAM 1500 5TH GEN"
+  RAM_HD = "RAM HD 5TH GEN"
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:25]: does this cover both the 2500 and 3500? if so, any reason to merge them into one here?

Comment (realfast) [reply selfdrive/car/chrysler/values.py:25]: Yes this covers Ram 2500-5500. This is why I labeled it HD vs 2500

File: selfdrive/car/torque_data/override.yaml (hunk 1 above)
Comment (realfast) [reply selfdrive/car/torque_data/override.yaml]: I forgot this was left in. I apologize.

File: selfdrive/car/chrysler/values.py [hunk 4]
```diff
@@ -177,6 +187,47 @@ class ChryslerCarInfo(CarInfo):
...
+    (Ecu.engine, 0x7e0, None): [
+      b'52421132AF',
+      b'M2370131MB',
+      b'M2421132MB',
+    ],
+    #(Ecu.transmission, 0x7e1, None): [
+      #b'68475153AD',
+      #b'68504640AC',
+      #b'68538041AB',
+    #],
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:228]: why is this commented out?

File: selfdrive/car/chrysler/values.py [hunk 5]
```diff
@@ -177,6 +187,47 @@ class ChryslerCarInfo(CarInfo):
       b'68500483AB',
     ],
   },
+
+  CAR.RAM_HD: {
+    (Ecu.combinationMeter, 0x742, None): [
+      b'68361606AH',
+      b'68492693AD',
+      b'68525485AB', #TODO: CHECK in newest updat on dongle 5c05760b592aee2c
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:198]: should I remove this?

File: selfdrive/car/chrysler/values.py (hunk 4 above)
Comment (realfast) [reply selfdrive/car/chrysler/values.py:228]: I do not feel the transmission is important for fingerprinting. Also on a truck with a 68rfe, the tcm logic is housed in the ecm. It’s considered a pcm. There will be no transmission fingerprint if I remember correctly.

File: selfdrive/car/chrysler/values.py (hunk 5 above)
Comment (realfast) [reply selfdrive/car/chrysler/values.py:198]: I do not feel it is needed for fingerprinting. So, it can be removed.

Comment (realfast) [reply selfdrive/car/chrysler/values.py:198]: I also, do not think gateway is needed on any of the chryslers

**Comment by jyoung8607** [comment]
Re: UDS fingerprinting, bear in mind the constellation of fingerprint modules needs to be:
* Widely, preferably universally shared among supported Chryslers
* Useful for distinguishing between different supported Chryslers, for selection of APIs and mass/wheelbase/tuning
* Useful for verifying lateral control API compatibility
* Useful for verifying longitudinal control API compatibility

Before discarding combinationMeter, do you have enough other info (from SRS or other addresses) to reliably tell a Jeep Grand Cherokee apart from a Pacifica apart from a Ram 1500 apart from a Ram HD? Are you confident that things like ESP and DASM firmware as-read-by this field aren't shared between models?

Before discarding transmission, do you know enough about the longitudinal control API that you can confidently do without knowing the trans type? Is the API acceleration based, or gas/brake percentage? Are there ever different transmission types available with the same chassis/engine that might affect gas tuning, or are there manual transmissions available?

File: selfdrive/car/tests/routes.py [hunk 6]
```diff
@@ -39,6 +39,7 @@
   TestRoute("8190c7275a24557b|2020-01-29--08-33-58", CHRYSLER.PACIFICA_2019_HYBRID),
   TestRoute("3d84727705fecd04|2021-05-25--08-38-56", CHRYSLER.PACIFICA_2020),
   TestRoute("221c253375af4ee9|2022-06-15--18-38-24", CHRYSLER.RAM_1500),
+  TestRoute("8fb5eabf914632ae|2022-08-04--17-28-53", CHRYSLER.RAM_HD, segment=6),
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/routes.py:42]: would also like to get a test route on this branch from a 3500

**Comment by RockaRhyme** [comment]
I just installed on my 2500. dongle id: 6038cd2b8addb4dc

will put in a few hundred miles tomorrow, but I'll be pulling a camper... not sure if that'll throw things off or not

**Comment by adeebshihadeh** [comment]
Ram HD will be "dashcam only" until #25389 is fixed.
